### PR TITLE
[3.9.x] [MNG-7473] Backport Maven BOM to 3.9.0

### PR DESCRIPTION
**Pull request by cstamas**
---

https://issues.apache.org/jira/browse/MNG-7473

### CODE REVIEW

File: maven-bom/pom.xml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    <site>
+      <id>apache.website</id>
+      <url>scm:svn:https://svn.apache.org/repos/asf/maven/website/components/${maven.site.path}</url>
+    </site>
+  </distributionManagement>
+
+  <dependencyManagement>
+    <!-- Not included:
+         - apache-maven, as it delivers the binaries
+         - maven-compat, only used by apache-maven distributions to support older releases
```
Comment (gnodet) [inline maven-bom/pom.xml:67]: I'm not sure it makes sense to not include `maven-compat`.  This is still required at runtime to run maven, so any plugin integration test will require it afaik.

File: maven-bom/pom.xml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    <downloadUrl>https://maven.apache.org/download.html</downloadUrl>
+    <site>
+      <id>apache.website</id>
+      <url>scm:svn:https://svn.apache.org/repos/asf/maven/website/components/${maven.site.path}</url>
+    </site>
+  </distributionManagement>
+
+  <dependencyManagement>
+    <!-- Not included:
+         - apache-maven, as it delivers the binaries
```
Comment (gnodet) [inline maven-bom/pom.xml:66]: Can't the distribution be used for ITs ?
  https://github.com/apache/maven-build-cache-extension/blob/master/pom.xml#L401-L409
Unfortunately, I don't think the `m-dependency-p` can leverage the info from the BOM, but I wonder if there are other use cases ?

**Review by gnodet — CHANGES_REQUESTED** [review]
Not sure about leaving out `apache-maven` and `maven-compat`...

**Review by gnodet — APPROVED** [review]
This is in line with current master, so out of topic for this PR.

**Review by slawekjaranowski — APPROVED** [review]

**Comment by kwin** [comment]
What about the comments from https://issues.apache.org/jira/browse/MNG-6806?focusedCommentId=17532971&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17532971.
This BOM is not really useful for plugin developers in its current form as putting the right version for those artifacts is a no-brainer. Can we have one which manages all artifacts exposed by the Core Classloader from the Maven distribution?

**Comment by cstamas** [comment]
@gnodet I also think leaving out maven-compat makes no sense (at least for 3.x line)
@kwin I completely agree, but was down voted for doing that by arguments "it is not BOM"...

In essence, I wanted to backport what master/4.x has, would merely add maven-compat and later we can see for the rest...

**Comment by kwin** [comment]
Maybe we can add another artifact then and call it maven-core-classloader-dep-mgmt. I don’t care about naming here, I primarily care about better support for plugin developers.

**Review by michael-o — APPROVED** [review]

**Comment by cstamas** [comment]
Am somewhat aligned with @kwin here, and am now on the other side: backporting the BOM from master to 3.9.x (as is) does not makes much sense (plus as said above, where is compat, where is maven etc), so am rather to close this out w/o merge. Will give 24h for anyone interested to respond (or just +1) this comment as sign for "I agree: close out this PR w/o merge".

**Comment by cstamas** [comment]
Am backing out from backporting BOM, it makes no sense as is IMHO

**Comment by jira-importer** [comment]
Resolve #9128

**Pull request closed without merging**